### ISSUE TITLE
Sync mode not saved correctly of disabled connection

### ISSUE THREAD

**Issue by timroes** [airbytehq/airbyte#15738]
The `v1/web_backend/connections/create` as well as `v1/web_backend/connections/updateNew` method (and potentially other connection updating methods?) are not saving the sync mode of a stream that has `selected: false` in its config. Instead further calls to get that connection will always return `full_refresh` + `append` for that stream.

This means that every connection that has disabled streams in it will when being edited be reset to full_refresh+append. This is not ideal, since the FE is at the moment the one handling the ideal default sync mode for a stream, which then might simply be overwritten by full_refresh+append for that stream if it's disabled.

This came up when watching several user session, where users went into connection to enable streams and all of them were suddenly at `full_refresh`+`append`.

cc @andyjih @evantahler

**Comment by alovew** [comment]
Are there steps to reproduce this? I tried updating a disabled stream's sync mode from full refresh + append to full refresh + overwrite and it worked.

**Comment by alovew** [comment]
Nvm, just managed to reproduce!

**Comment by evantahler** [comment]
Clarification questions from grooming:
* did this used to work?  Could you in the past update the sync mode of disabled streams, or did this never work?
* can we prevent modifying disabled streams in the frontend?  Is this a workflow we want?

**Comment by timroes** [comment]
>can we prevent modifying disabled streams in the frontend? Is this a workflow we want?

I think we can build this in theory. That said, given that the frontend is the one handling at the moment the default sync mode for connections, it feels we'll need to also apply that for disabled streams, to not end up in the scenario we do at the moment. Alternatively we should move the logic for optiomal sync modes into the backend instead, so that the backend is in charge of determining the right default sync mode for a stream. This would allow us to remove this code from the FE and at that point we could also just prevent editing disabled streams.... except that we're still doing more stuff in the FE by now (as discussed on the oncall yesterday), the FE is now also in charge to make sure the primary keys are matching the source defined primary keys, i.e. we'd also want to move that over to the backend then, if we'd no longer want to be able to modify disabled streams.

**Comment by timroes** [comment]
@evantahler I feel we should find an agreement first on whether the FE should be in charge of determining things like the optimal sync mode and setting primary keys correctly, or if we should move this over to the backend. Based on that (i.e. if the backend would take care of it), we can figure out if we want to prevent modifying disabled streams.

**Comment by lmossman** [comment]
It feels like this type of discussion (i.e. should the frontend or the backend contain X piece of logic around stream configurations?) has cropped up several times now. My general feeling here is that we should try to have as much of this logic as possible live in the backend. This has several benefits:
1. All of the business logic lives in one place (the backend). Since almost all of our business logic lives in the backend already, this just makes finding and fixing bugs like these easier for backend OC engineers, without requiring them to pull in anyone from the FE team for these small pieces of logic that are living in the frontend.
2. We have several OSS users who mainly interact with our product through our API, in which case any business logic that lives solely in the frontend will not be used. This means there is inconsistent behavior between the UI and the API, which is not great. Ideally the UI and the API are just different interfaces on our product which behaves the same way regardless of how it is being used.
3. Removing business logic from the frontend helps keep the frontend simpler, leading to fewer UI bugs and better performance

**Comment by evantahler** [comment]
I strongly agree with @lmossman.  What would it look like to audit all of the more complex logic and validation that the front-end does and come up with a holistic list of things we want to move into the backend?  This is probably an important step for making a robust Cloud API as well.

cc @salima-airbyte

**Comment by alovew** [comment]
This ticket requires some sort of doc or small tech spec with a plan for how to tackle this, with input from engineers & @andyjih. Issues to consider:
- catalog storage (configured catalogs on connection & source catalogs on actor_catalog)
- how to handle FE dealing with extra business logic
- how to handle connections that are completely disabled when their sync modes are updated & resets are required
